src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_CAPACITY = 10;
+    private T[] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[ARRAY_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        elements[size] = value;
+        size++;
+        if (elements.length == size) {
+            grow();
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (elements.length == size) {
+                grow();
+            }
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            elements[index] = value;
+            size++;
+        } else {
+            throw new ArrayIndexOutOfBoundsException();
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return elements[index];
+        }
+        throw new ArrayIndexOutOfBoundsException("Wrong index!");
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= 0 && index < size) {
```

Comment:
let's move the logic (if index for `get()`, `set()` and `remove()` methods is invalid) -> throw the exception
to the private method